### DISCUSSION TITLE
Enable kube-vip for control plane with BGP mode

### DISCUSSION THREAD

**Discussion by sathuish** [kube-vip/kube-vip#962] [Q&A]
Hi, I’m trying to enable kube-vip on the control plane with BGP mode, but the VIP is not reachable from worker nodes, which prevents me from executing any kubectl commands after adding it to the kube config. I’m also running Calico on my cluster, and I’m concerned about potential conflicts. Below is the command I’m using to generate the kube-vip manifests:

kube-vip manifest daemonset \
  --interface lo \
  --address 10.98.0.10 \
  --inCluster \
  --taint \
  --controlplane \
  --services \
  --bgp \
  --localAS 65000 \
  --bgpRouterID 192.168.0.3 \
  --k8sConfigPath=~/.kube/config \
  --bgppeers 192.168.0.1:65000::false,192.168.0.2:65000::false \
  --autodetectcp

Could you please advise on how to resolve this issue? Your help would be greatly appreciated!

**Comment by thebsdbox** [comment]
Can you get any logs from the kube-vip pods?

**Reply by sathuish** [reply]
[vip.zip](https://github.com/user-attachments/files/17359431/vip.zip)

Additionally, my clusters are set up on-premises, and the nodes are distributed across different global zones, each with distinct IP ranges, such as 192.168.xx, 172.15.xx, and 10.15.xx, etc..,. Which mode would be the best fit for this type of setup?
